src/Components/Admin/Rooms/roomModal.js

```diff
@@ -60,6 +73,31 @@ export default function RoomModal({
         );
     }, [room]);
 
+    const handleUploadFileChange = (event) => {
+        setErrors({ ...errors, assetUpload: "" });
+        if (event.target.files && event.target.files[0]) {
+            const file = event.target.files[0];
+            const name = file.name;
+            const lastDot = name.lastIndexOf(".");
+            const split = name.split(lastDot);
+            const fileName = split[0];
+            const fileType = split[1];
+
+            setFileName(fileName);
+            handleUploadAssetSubmit(fileName, fileType, file);
+        }
+    };
+
+    const handleUploadAssetSubmit = async (name, file_type, asset) => {
```

Comment:
Bad copypasta here too, the name should be fixed to reflect this is not an asset upload

Reply:
Sloppy. My bad!

Reply:
Correct me if I'm wrong, but I thought we generalized the upload functions so we don't need to keep creating these functions every time we need to upload to S3

Reply:
It is generalized! The `createPresignedLinkAndUploadS3` function does all the behind the scenes logic. The parent function just calls it and sets appropriate values on the response from this function call.